sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java

```diff
@@ -744,7 +758,12 @@ private void finishConfiguration(boolean configureFromApp) {
         mLogSerializer = new DefaultLogSerializer();
         mLogSerializer.addLogFactory(StartServiceLog.TYPE, new StartServiceLogFactory());
         mLogSerializer.addLogFactory(CustomPropertiesLog.TYPE, new CustomPropertiesLogFactory());
-        mChannel = new DefaultChannel(mApplication, mAppSecret, mLogSerializer, mHandler);
+        if (mAppSecret != null) {
+            mChannel = new DefaultChannel(mApplication, mAppSecret, mLogSerializer, mHandler);
+        } else {
+            mIngestion = new OneCollectorIngestion(mApplication, mLogSerializer);
```

Comment:
You must not change when One Collector is instantiated.

Instead you must keep a reference to the log url and apply setLogUrl when one collector is created, also remove the if and roll back this section of the code.